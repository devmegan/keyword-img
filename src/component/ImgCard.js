export default function ImgCard({ alt, src, setError, setLoading }){
    const handleLoad = () => {
        setLoading(false); 

        if(src.search("/source-404?") !== -1) {
            setError("No images found");
        }
    }

    return (
        <img 
            alt={alt} 
            className="card h-full w-auto mx-auto block object-cover shadow-2xl"
            src={src} 
            onLoad={handleLoad}
        />
    )
}