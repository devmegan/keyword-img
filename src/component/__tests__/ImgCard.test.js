import { render, screen, waitFor } from '@testing-library/react';
import { fireEvent } from '@testing-library/react';
import ImgCard from '../ImgCard';

const loadingFn = jest.fn();
const errorFn = jest.fn();

describe('img card', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  it('renders w/o crashing', () => {
    expect.hasAssertions();

    const view = render(<ImgCard />);
    
    expect(view).toMatchSnapshot();
  });

  it('handles loading status onload', async () => {
    expect.hasAssertions();
   
    render(<ImgCard alt="alt" setError={errorFn} setLoading={loadingFn} src="spaceship" />);

    const imgRef = screen.getByAltText('alt');

    fireEvent.load(imgRef);

    await waitFor(() => {
        expect(loadingFn).toHaveBeenCalledWith(false);
    });
  });

  it('handles image 404 error', async () => {
    expect.hasAssertions();

    render(<ImgCard alt="alt" setError={errorFn} setLoading={loadingFn} src="/source-404?" />);
    
    const imgRef = screen.getByAltText('alt');

    fireEvent.load(imgRef);

    await waitFor(() => {
        expect(errorFn).toHaveBeenCalledWith("No images found");
    });
  });
});