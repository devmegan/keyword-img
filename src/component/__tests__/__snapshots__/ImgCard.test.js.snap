// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`img card renders w/o crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        class="card h-full w-auto mx-auto block object-cover shadow-2xl"
      />
    </div>
  </body>,
  "container": <div>
    <img
      class="card h-full w-auto mx-auto block object-cover shadow-2xl"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
